src/tileLayer.js

```diff
@@ -275,6 +283,25 @@ var tileLayer = function (arg) {
     return $.extend({}, m_this._activeTiles); // copy on output
   }});
 
+  /**
+   * Get/set the queue object.
+   * @property {geo.fetchQueue} queue The current queue.
+   * @name geo.tileLayer#queue
+   */
+  Object.defineProperty(this, 'queue', {
+    get: function () { return m_this._queue; },
+    set: function (queue) {
+      if (m_this._queue !== queue) {
+        if (this._queue && this._queue._tileLayers && this._queue._tileLayers.indexOf(m_this) >= 0) {
```

Comment:
I have some idea why this is necessary but wondering if you could provide some explanations?

Reply:
There is a `needed` function that determines if a tile is still needed.  A tile is needed if it is needed by at least one layer that is using it.  The `_tileLayers` array is just a way to make sure we can enumerate the layers that are sharing the queue, and walk through the layers and ask if any layer needs a tile.  Setting the queue (or exiting the layer) needs to maintain the book keeping for this list of layers.

Reply:
I can add that as a comment.

Reply:
yeah that makes sense. So in the next line why are we setting this index to 1? The tileLayer is already has this queue before but the index may have been zero or above zero. Just wondering.

Reply:
We're not.  We are removing the layer from the old queue's _tileLayers list it was part of.  The splice(index, 1) says remove 1 value at its current location.  We then push in on the new queue's _tileLayers list at the end of the function.

Reply:
Ah thanks for reminding the function execution, okay that makes sense.